src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -307,10 +385,49 @@ public void setRead(final GATKRead read) {
                 }
             }
         }
+
+        // Returns true if either of the required mate information fields have been changed by the clipper
+        public boolean hasBeenOverhangClipped() {
+            return (!oldCigar.equals(read.getCigar())) || (oldStart != read.getStart());
+        }
+
+        // Adds the relevant information for repairing the mate to setMateChanged keyed on a string composed of the start position
+        public void setMateChanged() {
+            if (!read.isUnmapped()) {
```

Comment:
Good point! It occurs to me that for the potential future iteration of the tool though that assumption might not necessarily hold if we are overhang clipping supplementary alignments and need to save them for SA tags. One could potentially have a chimeric alignment that is unpaired.